web/components/charts/generic-charts.tsx

```diff
@@ -312,28 +319,46 @@ export const MultiValueHistoryChart = <P extends MultiPoint>(props: {
   )
 }
 
-export const SingleValueHistoryChart = <P extends HistoryPoint>(props: {
+export const ControllableSingleValueHistoryChart = <
+  P extends HistoryPoint
+>(props: {
   data: P[]
   w: number
   h: number
   color: string | ((p: P) => string)
   margin: Margin
   xScale: ScaleTime<number, number>
   yScale: ScaleContinuousNumeric<number, number>
+  viewXScale: ScaleTime<number, number, never> | undefined
+  setViewXScale: Dispatch<
+    SetStateAction<ScaleTime<number, number, never> | undefined>
+  >
+  viewYScale: ScaleContinuousNumeric<number, number, never> | undefined
+  setViewYScale: Dispatch<
+    SetStateAction<ScaleContinuousNumeric<number, number, never> | undefined>
```

Comment:
What do you think about bundling these 4 props within one object? e.g. `viewScaleProps`

You could export the type for this object from this file as well. Could make it easier to read when passed through other components.